xapian-letor/ranker/ranker.cc

```diff
@@ -351,19 +351,20 @@ Ranker::labelcomparer(const FeatureVector & firstfv, const FeatureVector& second
     return firstfv.get_label() > secondfv.get_label();
 }
 
-std::vector<Xapian::docid>
-Ranker::rank(const Xapian::MSet & mset, const string & model_key, const Xapian::FeatureList & flist)
+void
+Ranker::rank(Xapian::MSet & mset, const string & model_key, const Xapian::FeatureList & flist)
 {
     LOGCALL(API, std::vector<Xapian::docid>, "Ranker::rank", mset | model_key | flist);
     std::vector<FeatureVector> fvv = flist.create_feature_vectors(mset, letor_query, Xapian::Database(db_path));
     load_model_from_metadata(model_key);
     std::vector<FeatureVector> rankedfvv = rank_fvv(fvv);
 
-    std::vector<Xapian::docid> rankeddid;
+    std::vector<double> weights;
```

Comment:
This still creates a vector of doubles from a vector of things that contain those doubles. It should be possible to create an iterator over the vector which returns the letor score on `*`-deference, so it can be passed in directly to `set_new_weights()`. That said, it's getting into STL use that is quite subtle and definitely beyond what I've done in the past.